pkg/gen/input/makefile/internal/file/makefile.go

```diff
@@ -0,0 +1,82 @@
+package file
+
+import (
+	"github.com/giantswarm/devctl/pkg/gen/input"
+)
+
+func NewMakefileInput() input.Input {
+	i := input.Input{
+		Path:         "Makefile",
+		TemplateBody: makefileTemplate,
+		TemplateData: map[string]interface{}{},
+	}
+
+	return i
+}
+
+var makefileTemplate = `# DO NOT EDIT. Generated with:
+#
+#    devctl gen makefile
+#
+
+APPLICATION    ?= $(shell basename $(shell go list .))
+VERSION        ?= $(shell architect project version)
+GITSHA1        = $(shell git rev-parse --verify HEAD)
+BUILDTIMESTAMP = $(shell date -u '+%FT%TZ')
+LDFLAGS        ?= -w -linkmode 'auto' -extldflags '-static' \
+  -X '$(go list .)/pkg/project.buildTimestamp=${BUILDTIMESTAMP}' \
+  -X '$(go list .)/pkg/project.gitSHA=${GITSHA1}' \
+  -X '$(go list .)/pkg/project.version=${VERSION}'
+default: build
+
+.PHONY: build
+## build: builds a local binary
+build: clean
+	CGO_ENABLED=0 go build -o ${APPLICATION} .
+
+.PHONY: build-linux
+## build-linux: builds binary for linux/amd64
+build-linux: clean
+	GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build .
+
+.PHONY: build-darwin
+## build-darwin: builds binary for darwin/amd64
+build-darwin: clean
+	GOOS=darwin GOARCH=amd64 CGO_ENABLED=0 go build .
```

Comment:
```suggestion
.DEFAULT_GOAL := build

.PHONY: build build-darwin build-linux
## build: builds a local binary
build: $(APPLICATION)
## build-darwin: builds a local binary for darwin/amd64
build-darwin: $(APPLICATION)-darwin
## build-linux: builds a local binary for linux/amd64
build-linux: $(APPLICATION)-linux

$(APPLICATION): $(APPLICATION)-$(OS)
	cp -a $< $@

$(APPLICATION)-%: $(SOURCES)
	GOOS=$* GOARCH=amd64 go build -ldflags "$(LDFLAGS)" -o $@ .
```

Reply:
@kopiczko I tried that on my mac, `make build-linux` and `make build-darwin` just works but it doesn't work with `make build` or `make` directly.

```
cmd/go: unsupported GOOS/GOARCH pair .o/amd64
make: *** [devctl-.o] Error 2
```

Reply:
Can you paste the full outpu?